src/partitionedNLPModels/PBFGSNLPModels.jl

```diff
@@ -51,15 +53,16 @@ mutable struct PBFGSNLPModel{
   element_expr_tree_table::Vector{Vector{Int}} # length(element_expr_tree_table) == M
   index_element_tree::Vector{Int} # length(index_element_tree) == N, index_element_tree[i] ≤ M
 
-  vec_compiled_element_gradients::Vector{ReverseDiff.CompiledTape}
+  objective_backend::OB
+  gradient_backend::GB
 
-  op::P # partitioned quasi-Newton approximation
+  op::PartitionedStructures.Elemental_pm{T} # partitioned quasi-Newton approximation
 
   fx::T
   name::Symbol
 end
 
-function PBFGSNLPModel(nlp::SupportedNLPModel; type::DataType = Float64, merging::Bool = true)
+function PBFGSNLPModel(nlp::SupportedNLPModel; type::DataType = eltype(nlp.meta.x0), merging::Bool = true, kwargs...)
```

Comment:
```suggestion
function PBFGSNLPModel(nlp::SupportedNLPModel; type::Type{T} = eltype(nlp.meta.x0), merging::Bool = true, kwargs...) where {T}
```
and use `T` instead of `type` in the code

Reply:
Does that implies `Type{eltype(nlp.meta.x0)}`?